### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by whiteWolfx99**
In this group Capstone project, we created a Recipe app that keeps track of all our recipes, ingredients, and inventory.
1. we implemented these interfaces : 

- A login page.
- A registration page.
- A food list (with all CRUD implementation, except for 'update').
- A recipe list (with all CRUD implementation, except for 'update').
- A general shopping list view (all missing food for all your recipes and total price).
- A list of all public recipes from other users with their names and total prices.
- Inventories list (with all CRUD implementation, except for 'update').
- Inventory shopping list, a shopping list, but only taking into consideration a chosen recipe and inventory.

2. we write units and integration tests using Rspec and capybara

### this is how you can signup 

[video](https://drive.google.com/file/d/1Q6ecb26OXxATLzFT5xjYkmXlgo7RRlVS/view?usp=sharing)

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable, :confirmable
+  has_many :inventories
+  has_many :recipes
+end
```
Comment (AshakaE) [inline app/models/user.rb:8]: For some reason, I'm unable to create an account when signing up
image

- [ ] Kindly resolve this issue

File: spec/models/inventory_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+  it 'Food Name should be present' do
+    subject.Name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'Name should be inventory 1' do
+    expect(subject.Name).to eql 'inventory 1'
+  end
+end
```
Comment (AshakaE) [inline spec/models/inventory_spec.rb:15]: - [ ] Kindly work on the failing test cases in this spec
image

File: spec/views/food/foods_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      expect(page).to have_content('Add food')
+      expect(page).to have_content('Delete')
+    end
+
+    it 'button redirects to a page to add Add food' do
+      click_link 'Add food'
+      expect(page).to have_current_path new_food_path
+    end
+  end
+end
```
Comment (AshakaE) [inline spec/views/food/foods_index_spec.rb:39]: - [ ] Kindly work on the failing test cases in this spec 
image

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi @whiteWolfx99, @vanelnw  & @mohamed20163858 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Setting up your project locally wasn't difficult 👏 
- I appreciate the effort team members put into this project 👏 
- Proper use of Github flow ✅ 
- Pull request description ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (whiteWolfx99) [reply app/models/user.rb:8]: @AshakaE hi thanks for reviewing, i don't know it is working on my side, kindly look at this video is telling you how you can login.

[video](https://drive.google.com/file/d/1Q6ecb26OXxATLzFT5xjYkmXlgo7RRlVS/view?usp=sharing)

File: .gitignore [hunk 4]
```diff
@@ -4,6 +4,8 @@
 # or operating system, you probably want to add a global ignore instead:
```
Comment (MeqdamAlqudah) [inline .gitignore:4]: - [Opational] Please as the last reviewer said an error occur when the user try to sign up please fix it , although the new user can be created 
image

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @whiteWolfx99 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

image

## Highlights 
:+1: no linter errors 
:+1: the app is working and the design is cool 
image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**